head.S

```diff
@@ -67,8 +58,19 @@ GLOBAL(_entry)
 	 */
 	mov	%eax, %ebp
 
-	/* Set up the Stage 1 stack. */
-	lea	.L_stack_base(%ebp), %esp
+	/* Check if we have enough space available for stack. */
+	lea	bootloader_data(%ebp), %ebx
+	/* TODO: below uses bootloader_data.size, must be changed for SLRT. */
+	movzwl	%ss:2(%ebx), %edx
+	add	%edx, %ebx
+	add	$MAX_STACK_SIZE, %ebx
+	cmp	%ebx, %esp
+	jge	.Lenough_stack
```

Comment:
So you can actually use the fact that the SKL size is 64k to your advantage here:

```asm
movzwl (booloader_data-sl_header)+2(%ebp), %eax
add    $MAX_STACK_SIZE, %ax
jno    .Lenough_stack
```

ought to be equivalent, I think ?

Reply:
`booloader_data+2(%ebp)` has just the size from `booloader_data` to the end of that data, not from start of SLB, so `booloader_data` offset would still have to be added to that.

Using halfword trick is nice, but it won't matter for SLRT because it's size field is `u32`.

Reply:
Modified in https://github.com/TrenchBoot/secure-kernel-loader/pull/16/commits/65e9d9387c096cb55cac31035c01f2228f2a435f